cip/1.accepted/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc

```diff
@@ -164,13 +164,20 @@ In order to align equality with <<comparability-def>>, we change equality of lis
 Specifically, we propose to redefine how equality works for lists in Cypher.
 To determine if two lists `l1` and `l2` are equal, we propose two simple tests, as exemplified by the following:
 
-* `l1` and `l2` must have the same size, i.e. inversely `size(l1) <> size(l2) \=> l1 <> l2`
+* `l1` and `l2` must have the same size, i.e. inversely `size(l1) <> size(l2) \=> l1 <> l2`. This also applies when any of the lists contains null, i.e. `[a1, a2, ..., an] = [b1, b2, ..., null, ..., bm]` evaluates to false if `m <> n`.
 * the pairwise elements of both `l1` and `l2` must be equal, i.e.
 ----
 [a1, a2, ..., an] = [b1, b2, ..., bn]
 <=>
 a1 = b1 && a2 = b2 && ... && an = bn
 ----
+* If the second test evaluates to `null`, these lists are <<incomparable>>. This will only happen if at least one element in one of the lists is `null` while all non-null elements are pairwise equal. For example, `[1, 2] = [1, null]` evaluates to `null` (incomparable), while `[1, 2] = [null, 'foo']` evaluates to `false`.
```

Comment:
This contradicts the above example where `[1, 2] >= [1, 'text']` is said to be `null`, which proves that incomparability between lists is not at all dependent on an existence of `null` values within the lists.

Reply:
Also, why would `[1, 2] = [null, 'foo']` become `false`? It's `1 = null AND 2 = 'foo'` which is `null AND null` which is `null`, isn't it?